src/client/visualizations/scatterplot/scatterplot.tsx

```diff
@@ -103,27 +103,31 @@ export class Scatterplot extends React.Component<ChartProps, ScatterplotState> {
           ySeries={ySeries}/>}
         <GridLines orientation={"vertical"} stage={plottingStage} ticks={xTicks} scale={xScale}/>
         <GridLines orientation={"horizontal"} stage={plottingStage} ticks={yTicks} scale={yScale}/>
-        <XAxis scale={xScale} stage={calculateXAxisStage(plottingStage)} ticks={xTicks} formatter={xSeries.formatter()}
-               tickSize={TICK_SIZE}/>
+        <XAxis
+          scale={xScale}
+          stage={calculateXAxisStage(plottingStage)}
+          ticks={getTicksByDimension(xTicks, plottingStage.width)}
+          formatter={xSeries.formatter()}
+          tickSize={TICK_SIZE}/>
         <YAxis
           stage={calculateYAxisStage(plottingStage)}
-          ticks={yTicks}
+          ticks={getTicksByDimension(yTicks, plottingStage.height)}
```

Comment:
I don't like this name - we have very specific thing in our domain language that is called dimension. `getTicksForAvailableSpace`?

Reply:
Sure, that's much better, done ✅